buildSrc/src/main/groovy/org/elasticsearch/hadoop/gradle/BuildPlugin.groovy

```diff
@@ -365,12 +521,25 @@ class BuildPlugin implements Plugin<Project>  {
         // Co-locate all build artifacts into distributions subdir for easier build automation
         distribution.doLast {
             project.copy { CopySpec spec ->
```

Comment:
You should declare the inputs  / outputs of that task to allow gradle to detect it's up-to-date if nothing has changed. Probably better use a simple task of type `Copy` here